### DISCUSSION TITLE
Blazor United

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#2708] [General]
I am sure many of you have heard the recent news about Blazor United in .NET 8:

https://www.youtube.com/watch?v=48G_CEGXZZM
https://www.youtube.com/watch?v=kIH_Py8ytlY
https://www.dotnetrocks.com/details/1838

I am meeting with the Microsoft Blazor team tomorrow to discuss the opportunities for Oqtane to leverage this new capability.

**Comment by thabaum** [comment]
I look forward to hearing more about how Oqtane leverages .NET 8. I have been extremely excited for this next release.  More basic components like QuickGrid please!  Thanks for the update!

**Comment by sbwalker** [comment]
@thabaum the philosophy in Oqtane is that the framework should minimize dependencies, and developers should be able to utilize their preferred component library. So we do not intend to add any general components to the framework itself (except in those cases where a general component is needed to utilize framework services ie. FileManager, PermissionGrid). There is nothing stopping you from using the Blazor QuickGrid component in the current version of Oqtane.

**Comment by thabaum** [comment]
@sbwalker I was hoping you would put in a word for more created by the Blazor Team.  It would assist Oqtane without having to do anything here other than maybe suggest it to them.  

https://aspnet.github.io/quickgridsamples/

coming in .NET 8?

" QuickGrid is officially supported with .NET 8. "

This helps Oqtane developers minimize our dependencies.  It is a mission and if we can get these types of common UI components for charts, maps and others with most common basic features without having to import anything it would be cool :)

**Reply by sbwalker** [reply]
@thabaum I believe Microsoft's official stance on this is similar to Oqtane - they do not intend on providing a default UI component suite with Blazor as they want to encourage the ISV community to create many viable alternatives (it also keeps them focused on developing the framework without getting distracted with maintaining UI components). The QuickGrid POC is interesting because it slightly deviates from their official stance - so it will be interesting if they actually ship the component as part of Blazor, or maintain it separately.

**Reply by thabaum** [reply]
@sbwalker I do understand they are walking a fine line trying not to step on toes, however so many open source projects out there just trying to make the best of things, as long as it was open sourced who's toes are you stepping on?  The community would be able to build better on top of those and innovate from a base.  I feel like these should be handled more like dropping and dragging a form from VS 2000 or even 5.0 toolbox... maybe this would in a way actually be catching MS up with the times of the new components developers need ready in their toolbox.  Another view point I believe could be made that maybe the ISV community is holding innovation back as you have to adhere to new standards and development practices that vary more from the base.

I see the case both ways and I have been waiting for the .NET 8 discussion to arrive here.  I believe if they can now bring Maui into one project and have the UI experience native while styled... bingo.

**Reply by thabaum** [reply]
One last talking point I would like to make @sbwalker is having to trust all these other dependencies. I already trust MS by using .NET Core.  Seems you will be discussing things with them as a member of the Oqtane community maybe some of these talking points may be worthy of noting in discussion.  Thank you!

**Comment by thabaum** [comment]
Anything cool come from the meeting by chance?  

Any news or feedback for the Oqtane Framework vision that should help?  

Maybe the chat channel is filled with info, but I generally just get the feeds here so any news when you have a moment looking forward to hearing.  I am hoping my suggestion for a new template gets some love with just a blank index page, no components or code, but the layouts, nav and theme basics :).   I find one template too little the other too distracting.  Looking for a happy medium. Blazor team said they put it on the back burner to give it more thought. 

 Sounds like .NET 8 is planned for release in November at the .NET Conf.  Will we be creating Oqtane a .NET 7 update or focus more on .NET 8 with Oqtane?

Any news on any date for .NET 8 changing or do things look like they are going to plan?

I will try to assist as I can, I have a new project I am looking to build on Oqtane.  This will be my first production site using Oqtane I plan.

It looks like a few of features we should just be able to add a line of code to give the experience created when you create virtual lazy loading.

Are there any breaking changes we need to see if we can iron out to move forward by the time it is release?  .NET 8 LTS so seems like a goal to be ready for it.

Sorry so many questions I want to use Oqtane, but I can build from ground up as well and not wait for community to catch up on some projects.  I am invested here already here so when you get a moment hit me up with what you can.  I just got done learning probably 5 different approaches to the same engineering issues... 

I have come up with my own favorite structure and Oqtane is close but not 100% in which how could it be to my liking... now with all that said it is your approach as founder of the project and not a bad one although I am sure there are one or two things wish could of been done differently at the time of conception. .NET Core Blazor has changed a lot and Oqtane Framework is one of pioneers in using this technology so this is expected.

Thanks again, this project has been fun!  I wish I could help more.

**Comment by Sen-Gupta** [comment]
Hello @sbwalker ,

Here is my take:

1) For Oqtane most of the pages may be blog/content only, they will benefit from blazor united server side rendering, Oqtane already does that using controllers/PageModel
2) Capability to seamlessly convert to progressive web application/or render a components from serverstreaming in case page has a component that needs user inputs and a lots of server side validation with secrets or protected apis behind firewalls
3) Capability to go totally client side for things like, a simple interest calculator/a rich Data Entry Form, not too many calls for validations or security issues

The most important aspect is that all the model of page rendering work seamlessly. (Let's hope)

What I'm not sure, if these are supported across MAUI as well? (Haven't heard on it)
(Try printing on MAUI cross platform for fun)

There are lots of use cases and it seems Microsoft is lately trying to bring back some sense into future projects.
(Been working on MS Stack for my entire life but hate to see that when I have to do simple microservice, the developer life cycle in some other stacks is way faster yielding a superb go to market time)

Regards

**Comment by sbwalker** [comment]
The initial conversation with Microsoft was focused on explaining the current capabilities of Oqtane... which is important because Oqtane really pushes the limits of Blazor and contains a lot of very advanced concepts and functionality. A standard Blazor application is single-tenant, monolithic, and tightly coupled... whereas Oqtane is multi-tenant, modular/composable, and loosely coupled. And the challenge is that if Microsoft is only considering the simple use case when developing Blazor United, it may result in more advanced applications such as Oqtane not being able to leverage the new features, or even worse, breaking changes which affect Oqtane's ability to be successful. So the initial meeting was purely educational and there are a variety of topics which I will need to follow up on with the Blazor team to ensure there is clear understanding of the technical details.
